clarifications to libp2p requirements and block validation (#284)
filecoin-project/specs: INTRO.md

### Message Transport

> Filecoin uses [libp2p](https://libp2p.io) for all network communications. libp2p provides transport-agnostic services for peer discovery, naming, routing, pubsub channels and a distributed record store, and there are full or partial [implementations](https://libp2p.io/implementations/) in a number of languages. This spec assumes the use of libp2p and its services and does not specify transport-level details. That said, to be at least minimally compatible with other Filecoin nodes, we recommend supporting at least the [mplex](https://github.com/libp2p/specs/tree/master/mplex) stream multiplexer, and the [secio](TODO, spec in PR) encrypted transport protocols. For more details on the exact wire protocol of libp2p, refer to the [libp2p specs](https://github.com/libp2p/specs).

Reviewer: more spec-speek maybe? `-we recommend supporting at least`  `+it should support at least`

```suggestion
Filecoin uses [libp2p](https://libp2p.io) for all network communications. libp2p provides transport-agnostic services for peer discovery, naming, routing, pubsub channels and a distributed record store, and there are full or partial [implementations](https://libp2p.io/implementations/) in a number of languages. This spec assumes the use of libp2p and its services and does not specify transport-level details. That said, to be at least minimally compatible with other Filecoin nodes, it should support at least the [mplex](https://github.com/libp2p/specs/tree/master/mplex) stream multiplexer, and the [secio](TODO, spec in PR) encrypted transport protocols. For more details on the exact wire protocol of libp2p, refer to the [libp2p specs](https://github.com/libp2p/specs).
```

or maybe even `must`?

```suggestion
Filecoin uses [libp2p](https://libp2p.io) for all network communications. libp2p provides transport-agnostic services for peer discovery, naming, routing, pubsub channels and a distributed record store, and there are full or partial [implementations](https://libp2p.io/implementations/) in a number of languages. This spec assumes the use of libp2p and its services and does not specify transport-level details. That said, to be at least minimally compatible with other Filecoin nodes, it must support at least the [mplex](https://github.com/libp2p/specs/tree/master/mplex) stream multiplexer, and the [secio](TODO, spec in PR) encrypted transport protocols. For more details on the exact wire protocol of libp2p, refer to the [libp2p specs](https://github.com/libp2p/specs).
```

> Filecoin uses [libp2p](https://libp2p.io) for all network communications. libp2p provides transport-agnostic services for peer discovery, naming, routing, pubsub channels and a distributed record store, and there are full or partial [implementations](https://libp2p.io/implementations/) in a number of languages. This spec assumes the use of libp2p and its services and does not specify transport-level details. That said, to be at least minimally compatible with other Filecoin nodes, we recommend supporting at least the [mplex](https://github.com/libp2p/specs/tree/master/mplex) stream multiplexer, and the [secio](TODO, spec in PR) encrypted transport protocols. For more details on the exact wire protocol of libp2p, refer to the [libp2p specs](https://github.com/libp2p/specs).

Reviewer 2: maybe add another todo/note that we would like to switch to TLS 1.3 if it becomes ready/stable?

Author: eh, I don't think thats necessary. If we switch, we switch and update it in the spec. If we don't, we don't. There are enough random TODOs laying around
